From 9252fc4043595391f7c007dd1689d6bfbfafec6a Mon Sep 17 00:00:00 2001
From: Emmanuel Berthier <emmanuel.berthier@intel.com>
Date: Thu, 20 Aug 2015 16:19:52 +0200
Subject: [PATCH 27/41] [LOCKDEP] Correct some wrong lock initializations

We need to use rwlock_init() instead of __RW_LOCK_UNLOCKED() in i915_gem_userdata()

We need to initialize the obj_attr before calling sysfs_create_bin_file()
in i915_gem_create_sysfs_file_entry()

Change-Id: I8a65e7d44fcda06fbfaf8c6f20e3ed13ba83af3a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31683
Signed-off-by: Emmanuel Berthier <emmanuel.berthier@intel.com>
Reviewed-on: https://android.intel.com:443/493123
Reviewed-on: https://android.intel.com:443/504765
---
 drivers/gpu/drm/i915/i915_gem_userdata.c | 2 +-
 drivers/gpu/drm/i915/i915_sysfs.c        | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_userdata.c b/drivers/gpu/drm/i915/i915_gem_userdata.c
index c670ffd..e3cae14 100644
--- a/drivers/gpu/drm/i915/i915_gem_userdata.c
+++ b/drivers/gpu/drm/i915/i915_gem_userdata.c
@@ -110,7 +110,7 @@ i915_gem_userdata(struct drm_device *dev,
 
 		userdata_blk->length = (u16)bytes;
 		userdata_blk->flags  = flags;
-		userdata_blk->lock = __RW_LOCK_UNLOCKED(userdata_blk->lock);
+		rwlock_init(&userdata_blk->lock);
 
 		if (data) {
 			ret = copy_from_user(userdata_blk->data, data, bytes);
diff --git a/drivers/gpu/drm/i915/i915_sysfs.c b/drivers/gpu/drm/i915/i915_sysfs.c
index d66653c..129f5b9 100644
--- a/drivers/gpu/drm/i915/i915_sysfs.c
+++ b/drivers/gpu/drm/i915/i915_sysfs.c
@@ -1708,6 +1708,7 @@ int i915_gem_create_sysfs_file_entry(struct drm_device *dev,
 	}
 
 	snprintf(attr_priv->tgid_str, 16, "%d", task_tgid_nr(current));
+	sysfs_bin_attr_init(obj_attr);
 	obj_attr->attr.name = attr_priv->tgid_str;
 	obj_attr->attr.mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
 	obj_attr->size = 0;
-- 
1.9.1

